### PR TITLE
Catalog Item: Ensure it passes strings for deps.

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: cfme/tests/services/test_service_catalogs.py -v -k no_template_catalog_item --use-provider vsphere55}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 24fbec90d8e439f2f1de85fab5c86b9f94092c5d:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/services/catalogs/catalog_item.py [hunk 1]
```diff
@@ -208,8 +208,8 @@ def create(self):
         fill(basic_info_form, {'name_text': self.name,
                                'description_text': self.description,
                                'display_checkbox': self.display_in,
-                               'select_catalog': self.catalog,
-                               'select_dialog': self.dialog,
+                               'select_catalog': str(self.catalog),
```
Comment (psav) [inline cfme/services/catalogs/catalog_item.py]: I guess these should actually use the ``.name`` property?

Comment (mfalesni) [inline cfme/services/catalogs/catalog_item.py]: I guess we want to lean towards the option of passing objects.

Comment (psav) [inline cfme/services/catalogs/catalog_item.py]: Yes, but I'll merge this for now

**Pull request merged**